generate.js

```diff
@@ -65,31 +66,35 @@ function processIncludes(input, cb) {
   console.error(includes);
   var incCount = includes.length;
   if (incCount === 0) cb(null, input);
+
   includes.forEach(function(include) {
     var fname = include.replace(/^@include\s+/, '');
-    if (!fname.match(/\.md$/)) fname += '.md';
-
-    if (includeData.hasOwnProperty(fname)) {
-      input = input.split(include).join(includeData[fname]);
-      incCount--;
-      if (incCount === 0) {
-        return cb(null, input);
-      }
-    }
-
-    var fullFname = path.resolve(path.dirname(inputFile), fname);
-    fs.readFile(fullFname, 'utf8', function(er, inc) {
-      if (errState) return;
-      if (er) return cb(errState = er);
-      processIncludes(inc, function(er, inc) {
-        if (errState) return;
-        if (er) return cb(errState = er);
+    getFileEnding(path.resolve(path.dirname(inputFile), fname), function(err, ending) {
+      if (err) throw err;
```

Comment:
i will create an error handler which logs in a separate PR, currently i would like to follow the `throw` that is used in the whole codebase